clients/src/main/java/org/apache/kafka/common/errors/UnsupportedVersionException.java

```diff
@@ -16,6 +16,11 @@
  */
 package org.apache.kafka.common.errors;
 
+/**
+ * Indicates that a request API or version needed by the client is not supported by the broker. This is
+ * typically a fatal error as Kafka clients will downgrade request versions as needed except in cases where
```

Comment:
To elaborate a little more, if you use a consumer method that returns `UnsupportedVersionException`, you may be able to fallback to an alternative (say `offsetForTimes`). Do we consider that `fatal`?